plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -195,11 +195,11 @@ instance HasRenderContext config => PrettyBy config (DefaultUni a) where
         DefaultUniApply uniF uniA -> uniF `juxtPrettyM` uniA
         DefaultUniData            -> "data"
 
+instance HasRenderContext config => PrettyBy config (SomeTypeIn DefaultUni) where
+    prettyBy config (SomeTypeIn uni) = prettyBy config uni
+
 -- | This always pretty-prints parens around type applications (e.g. @(list bool)@) and
 -- doesn't pretty-print them otherwise (e.g. @integer@).
--- This is so we can have a single instance that is safe to use with both the classic and the
--- readable pretty-printers, even though for the latter it may result in redundant parens being
--- shown. We are planning to change the classic syntax to remove this silliness.
 instance Pretty (DefaultUni a) where
```

Comment:
Well, `Pretty` is for classic pretty-printing and `PrettyBy` is for readable one. We render built-in functions the same way in both, so we use `Pretty` for both, and we used to render built-in types the same way, but now they're just a tiny bit different. If we were to unify the two, then I think we would have to make the classic pretty-printer dependent on the readable one, so that we can call an equivalent of

```haskell
prettyBy $ RenderContext ToTheRight juxtFixity
```

(the current implementation of `Pretty (DefaultUni a)`) in the classic pretty-printer. But then both classic and readable pretty-printers carry a `configName` and we'd have to basically convert the config for the classic pretty-printer into the config of the readable one. Which doesn't seem too hard, but I don't feel comfortable entangling the two pretty-printers that much.

... on the other hand if we made the constraint `PrettyBy RenderContext` in both the cases and

1. created a `RenderContext` on the fly in the classic pretty-printer
2. extracted the `RenderContext` from the current config in the readable pretty-printer 

then it all would actually work, it seems.

Thanks for nagging me about it! I'm going to try out that latter idea.

Reply:
I think that's kind of what I expected. In my head it's something like:
```
fooWithArg arg = ....
foo = fooWithArg def

Reply:
Yeah, I'm just being stupid.

Reply:
The `Pretty` instances have to stay, because we do occasionally need them when fitting universes stuff into an function with a `Pretty a` constraint (in particular, that happens in the QuviQ's unification machinery). But all duplication is gone, which is good.